### PR TITLE
feat: update react/displayName, react/sortComp and indentation 

### PR DESCRIPTION
**Pull request by alihassan0**
BREAKING CHANGE
2 spaces indentation with prettier 
Screenshot from 2019-05-19 11-54-04

current 4 spaces indentation   
Screenshot from 2019-05-19 11-54-25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,33 @@ var HelloJohn = createReactClass({
 
 1.3 **Do not** use `React.createElement` unless you’re initializing the app from a file that is not JSX.
 
+1.4 **Always** use a displayName for components when exporting them so they can be easily identified in react dev tools
```
Comment (bahaagalal) [inline README.md]: Move this rule to **section 3: Naming**. So it becomes 3.5.

File: README.md [hunk 2]
```diff
@@ -46,6 +46,33 @@ var HelloJohn = createReactClass({
 
 1.3 **Do not** use `React.createElement` unless you’re initializing the app from a file that is not JSX.
 
+1.4 **Always** use a displayName for components when exporting them so they can be easily identified in react dev tools 
+```javascript
+// Bad
+export default () => {
+  return <button />;
+}
+// Good
```
Comment (bahaagalal) [inline README.md]: leave a newline before `// Good`

File: README.md [hunk 3]
```diff
@@ -46,6 +46,33 @@ var HelloJohn = createReactClass({
...
+// Named export (declaration)
+export function Button() {
+  return <button />;
+}
+
+//Named export (arrow)
```
Comment (bahaagalal) [inline README.md]: We prefer leaving a space after `//` in comments. 

Apply this to all comments here.

File: README.md [hunk 4]
```diff
@@ -46,6 +46,33 @@ var HelloJohn = createReactClass({
 
 1.3 **Do not** use `React.createElement` unless you’re initializing the app from a file that is not JSX.
 
+1.4 **Always** use a displayName for components when exporting them so they can be easily identified in react dev tools 
+```javascript
+// Bad
```
Comment (bahaagalal) [inline README.md]: we prefer using small caps with all comments so this should become `// bad` 

Apply this to all code comments here.

File: README.md [hunk 5]
```diff
@@ -446,16 +473,6 @@ const styles = { color: "red" };
...
-   
-```javascript
-// bad
-var Hello = <Hello personal />;
-
-// good
-var Hello = <Hello personal={true} />;
-```
-
 6.8 Ensures that any component or prop methods used to handle events are correctly prefixed. ([jsx-handler-names](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md))
```
Comment (bahaagalal) [inline README.md]: If the above rule is ended up being removed, you need to adjust the numbering so this shall become `6.7` and the next one shall become `6.8`

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-crowdanalyzer-react",
-  "version": "1.2.0",
+  "version": "2.2.0",
```
Comment (bahaagalal) [inline package.json]: When incrementing a major version, you zero both patch and minor version. So this should become `2.0.0`

File: README.md [hunk 7]
```diff
@@ -446,16 +473,6 @@ const styles = { color: "red" };
 <div style={styles} />
 ```
 
-6.7 Avoid passing a boolean attribute in jsx without specifying its value. ([jsx-boolean-value](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md))
```
Comment (bahaagalal) [inline README.md]: Why deleting this rule? Also if you intend to remove it, you didn't actually remove it from eslint rules in `jsx` file.

File: README.md (hunk 1 above)
Comment (alihassan0) [reply README.md]: I believe this has more to how to export a component than how to name them

Comment (bahaagalal) [reply README.md]: @alihassan0 agreed

File: rules/jsx.js [hunk 8]
```diff
@@ -187,7 +179,7 @@ module.exports = {
          * <React.Fragment>...</React.Fragment>
          * https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-fragments.md
          */
```
Comment (bahaagalal) [inline rules/jsx.js]: update `This rule allows you to enforce <React.Fragment>...</React.Fragment>` to `This rule allows you to enforce shorthand syntax`

**Comment by bahaagalal** [comment]
@alihassan0 Update the indentation to 2 spaces. Make sure you update both the rule comment in the rule file as well as the Readme file to reflect the change.

**Pull request merged**